packages/autocomplete-html/update/update.js

```diff
@@ -0,0 +1,289 @@
+  };
+
+  // Now to write our file
+  fs.writeFileSync("./completions.json", JSON.stringify(completion, null, 2));
+
+  console.log("Updated all `autocomplete-html` completions.");
+}
+
+function buildHtmlElementsArray(elements) {
+  let elementArray = [];
+
+  for (const spec in elements) {
+    if (Array.isArray(elements[spec].elements)) {
+      for (const ele of elements[spec].elements) {
+        elementArray.push(ele);
+      }
+    }
+  }
+
+  return elementArray;
+}
+
+function matchElementInterface(elements, domProperties) {
+  let outElements = {};
+
+  for (const ele of elements) {
+    let tags = resolveElementInterfaceAttrs(ele, domProperties);
+
+    outElements[ele.name] = {};
+
+    if (tags.length > 0) {
+      outElements[ele.name].attributes = tags;
+    }
+
+    let desc = getElementDescription(ele.name);
+    outElements[ele.name].description = desc;
+
+  }
+
+  return outElements;
+}
+
+function resolveElementInterfaceAttrs(element, domProperties) {
+  let attrs = [];
+  let interfaceArray = [];
+
+  if (typeof element.interface === "string") {
+    interfaceArray.push(element.interface);
+  }
+
+  // Now to loop through every interface and resolve the tags within.
+  while (interfaceArray.length > 0) {
+    let interface = interfaceArray[0];
+
+    if (domProperties[interface]) {
+      // First add all immediate props
+      for (const prop in domProperties[interface].props) {
+        attrs.push(prop);
+
+        // Then add any needed values to our global attributes
+        if (domProperties[interface].props[prop].global && !GLOBAL_ATTRIBUTES.includes(prop)) {
+          GLOBAL_ATTRIBUTES.push(prop);
+        }
+      }
+      // Now resolve any additional interfaces, by adding them to our existing array
+      if (typeof domProperties[interface].inheritance === "string") {
+        interfaceArray.push(domProperties[interface].inheritance);
+      }
+      if (Array.isArray(domProperties[interface].includes)) {
+        interfaceArray = interfaceArray.concat(domProperties[interface].includes);
+      }
+    }
+    // Now we have done everything needed for this one interface to be resolved,
+    // so we can just remove the first element of the array and let the while
+    // loop continue
+    interfaceArray.shift();
+  }
+
+  // Return our final list of attributes
+  return attrs;
+}
+
+function getElementDescription(element) {
+  // We will gather a description by checking if there's a document written
+  // on MDN for our Element and then extract a summary from there.
+
+  // Some elements are similar enough they exist in a single folder of a different
+  // name. So we will manually intervein in those cases.
+
+  const substitueElements = {
+    "h1": "heading_elements",
+    "h2": "heading_elements",
+    "h3": "heading_elements",
+    "h4": "heading_elements",
+    "h5": "heading_elements",
+    "h6": "heading_elements"
+  };
+
+  if (substitueElements[element]) {
+    element = substitueElements[element];
+  }
```

Comment:
I feel we can make this easier by just using a regexp:

```suggestion
  if (element.match(/^h[1-6]$/)) {
    element = 'heading_elements';
  }
```

Reply:
Yeah good point. I originally wrote it not knowing how many elements would need substitutions, but this looks good to me, thanks!